### PR TITLE
VMware/vSphere virtualization support for CSI Driver

### PR DESCRIPTION
**Pull request by delldubey**
# Description
- Enable VMware virtualized environment support (via RDM)
- Attach an RDM to a host
- Detach an RDM from a host
- Adds a sample storage class to be used for vSphere volumes

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/528|

# Checklist:

- [x] Have you run format,vet & lint checks against your submission?
- [x] Have you made sure that the code compiles?
- [x] Did you run the unit & integration tests successfully?
- [ ] Have you maintained at least 90% code coverage?
- [ ] Have you commented your code, particularly in hard-to-understand areas
- [x] Have you done corresponding changes to the documentation
- [x] Did you run tests in a real Kubernetes cluster?
- [x] Backward compatibility is not broken

# How Has This Been Tested?
- Unit test
 544 scenarios (544 passed)
3865 steps (3865 passed)
4m11.113514212s
godog finished

### CODE REVIEW

**Comment by shanmydell** [comment]
@delldubey : Please resolve formatting and gosec issues

File: helm/csi-powermax/templates/controller.yaml [hunk 1]
```diff
@@ -402,6 +402,20 @@ spec:
               value: "{{ .Values.controller.healthMonitor.enabled }}"
             {{- end }}
             {{- end }}
+            - name: X_CSI_VSPHERE_ENABLED
+              value: "{{ .Values.vSphere.enabled }}"
+            {{- if eq .Values.vSphere.enabled true }}
+            - name: X_CSI_VSPHERE_PORTGROUP
+              value: {{ required "Must provide portgroup for vsphere" .Values.vSphere.fcPortGroup }}
+            - name: X_CSI_VSPHERE_HOSTGROUP
+              value: {{ required "Must provide portgroup for vsphere" .Values.vSphere.fcHostGroup }}
```
Comment (shanmydell) [inline helm/csi-powermax/templates/controller.yaml:411]: required message contains same value for all attributes

**Review by shanmydell — CHANGES_REQUESTED** [review]

File: helm/csi-powermax/templates/node.yaml [hunk 2]
```diff
@@ -177,6 +177,20 @@ spec:
               value: "{{ .Values.node.topologyControl.enabled }}"
             {{- end }}
             {{- end }}
+            - name: X_CSI_VSPHERE_ENABLED
+              value: "{{ .Values.vSphere.enabled }}"
+            {{- if eq .Values.vSphere.enabled true }}
+            - name: X_CSI_VSPHERE_PORTGROUP
+              value: {{ required "Must provide portgroup for vsphere" .Values.vSphere.fcPortGroup }}
```
Comment (shanmydell) [inline helm/csi-powermax/templates/node.yaml:184]: Same comment. update the required message

Comment (boyamurthy) [reply helm/csi-powermax/templates/node.yaml:184]: updated message.

File: helm/csi-powermax/templates/controller.yaml (hunk 1 above)
Comment (boyamurthy) [reply helm/csi-powermax/templates/controller.yaml]: updated message .

**Review by shanmydell — APPROVED** [review]

**Review by boyamurthy — APPROVED** [review]
LGTM.

**Pull request merged**